### ISSUE TITLE
Distributed autograd tests are flaky under spawn mode

### ISSUE THREAD

**Issue by mrshenli** [pytorch/pytorch#27157]
https://circleci.com/gh/pytorch/pytorch/3023279?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

cc @pietern @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini

**Comment by mrshenli** [comment]
Tests disabled in #27158

**Comment by pritamdamania87** [comment]
@mrshenli @zhaojuanmao Would this be fixed by https://github.com/pytorch/pytorch/pull/27191 ?

**Comment by pritamdamania87** [comment]
Looks like the issue is similar to the other tests that we saw that were failing on py3.5:

```
#0  0x000055c2995f9cd4 in code_dealloc ()
#1  0x000055c2995fa538 in func_dealloc ()
#2  0x00007f7a216cdbcf in torch::distributed::rpc::PythonRpcHandler::~PythonRpcHandler() () from /home/pritam/local/anaconda3/envs/pytorch_py35/lib/python3.5/site-packages/torch/lib/libtorch_python.so
#3  0x00007f7a2be19b69 in __run_exit_handlers () from /lib64/libc.so.6
#4  0x00007f7a2be19bb7 in exit () from /lib64/libc.so.6
#5  0x000055c2996984d1 in Py_Exit ()
#6  0x000055c2996985b8 in handle_system_exit ()
#7  0x000055c299698622 in PyErr_PrintEx ()
#8  0x000055c2996a8806 in PyRun_SimpleStringFlags ()
#9  0x000055c2996a9679 in Py_Main ()
#10 0x000055c299573571 in main ()
```

**Comment by zhaojuanmao** [comment]
yes, it will be fixed by  #27191  temporarily
